File: python/questions/longestBalanced2.py
class Solution:
    def longestBalanced(self, s: str) -> int:
        ans = 0
        ss = len(s)
        for l in range(ss):
            cnt = defaultdict(int)
            for r in range(l, ss):
                cnt[s[r]] += 1
                if len(set(cnt.values())) == 1:
                    ans = max(ans, r-l+1)
        return ans

File: cpp/questions/longestBalanced2.cpp
class Solution {
public:
    int longestBalanced(string s) {
        int ans = 0;
        int ss = s.size();
        for (int l=0; l<ss-ans; l++) {
            int cnt[26]{};
            for (int r=l; r<ss; r++) {
                int cc = ++cnt[s[r]-'a'];
                bool valid = true;
                for (int c: cnt) {
                    if (c && c != cc) {
                        valid = false;
                        break;
                    }
                }
                if (valid) {
                    ans = std::max(ans, r-l+1);
                }
            }
        } 
        return ans;
    }
};
